runtime/query.go

```diff
@@ -101,18 +98,6 @@ func (r *Runtime) Query(ctx context.Context, instanceID string, query Query, pri
 		return query.Resolve(ctx, r, instanceID, priority)
 	}
 
-	// Skip caching if the OLAP connector is not DuckDB.
-	// NOTE: This hack is removed in the new resolvers by letting the resolver itself decide whether to cache or not.
-	olap, release, err := r.OLAP(ctx, instanceID, olapConnector)
-	if err != nil {
-		return err
-	}
-	dialect := olap.Dialect()
-	release()
-	if dialect != drivers.DialectDuckDB {
-		return query.Resolve(ctx, r, instanceID, priority)
-	}
```

Comment:
This will make sure profiling queries are cached for clickhouse as well which is what we want right ? And the results will get invalidated if source/model change similar to how it happens in duckdb.

Reply:
I don't really know – what if someone refreshes their underlying data and then don't understand why Rill Developer doesn't reflect the changes?

Maybe caching profiling for a few mins would be good, but waiting until the next restart sounds like it might be too confusing.

Reply:
Okay my understanding was that for rill managed tables (source and models) users should not add data externally.
I can revert to original implementation as well.

Reply:
Maybe I misunderstood the context here – I was thinking about profiling on user-managed tables, not Rill-managed tables.

I think the UI may still run profiling on user-managed tables in some cases, but I might be wrong. Anyway they might start doing it, so it would be nice not to have a tricky-to-debug cache issue surface if they do.

Reply:
Okay so for external tables the following check(already present) will make sure caching is disabled :
```
// If there were no known dependencies, skip caching
if len(depKeys) == 0 {
	return query.Resolve(ctx, r, instanceID, priority)
}
```
For models and sources it will be good to cache results similar to duckdb.
What do you think ?

Reply:
Makes sense, yeah that check seems sufficient!